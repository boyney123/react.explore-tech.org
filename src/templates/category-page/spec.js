import React from 'react'
import { shallow } from 'enzyme'

import Template from './'

const graphResult = {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          fields: {
            slug: '/random',
          },
          frontmatter: {
            path: '/',
            img: {},
            material: {
              author: {
                name: 'David',
              },
              title: 'Random',
              subtitle: 'Random item on the website',
              url: 'https://www.test.co.uk',
              tags: ['random', 'test'],
              stargazers_count: 1000,
              subscribers_count: 1000,
              github_url: 'https://www.github.com/boyney123/testing',
            },
          },
        },
      },
      {
        node: {
          fields: {
            slug: '/random-2',
          },
          frontmatter: {
            path: '/random-2',
            img: {},
            material: {
              author: {
                name: 'David',
              },
              title: 'New Title 2',
              subtitle: 'Another Random item on the website',
              url: 'https://www.test-2.co.uk',
              tags: ['random', 'test'],
              stargazers_count: 1000,
              subscribers_count: 1000,
              github_url: 'https://www.github.com/boyney123/testing',
            },
          },
        },
      },
    ],
  },
}

const pageContext = {
  category: 'random',
  currentPage: 2,
  numPages: 3
}

describe('Template', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Template data={graphResult} pageContext={pageContext} />)
    expect(wrapper).toMatchSnapshot()
  })
})
