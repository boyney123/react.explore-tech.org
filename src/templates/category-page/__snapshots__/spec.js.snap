// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template matches the snapshot 1`] = `
<Layout
  className="blog-post-container"
  header={
    <Header
      contribute={true}
      subtitle="react.explore-tech.org"
      title="random"
    />
  }
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="react.explore-tech.org"
  >
    <script
      async={true}
      charset="utf-8"
      src="https://platform.twitter.com/widgets.js"
    />
    <link
      crossorigin="anonymous"
      href="https://use.fontawesome.com/releases/v5.6.1/css/all.css"
      integrity="sha384-gfdkjb5BdAXd+lj+gudLWI+BXq4IuLW5IT+brZEZsLFm++aCMlF1V92rMkPaX4PP"
      rel="stylesheet"
    />
    <html
      lang="en"
    />
  </HelmetWrapper>
  <div
    className="container"
  >
    <div
      className="columns"
    >
      <div
        className="column is-one-quarter category-side-bar"
      >
        <input
          className="category-side-bar__search"
          onChange={[Function]}
          placeHolder="Filter by name, tags, etc..."
        />
        <h5
          className="category-side-bar__tags__title mt20"
        >
          Tags
        </h5>
        <ul
          className="category-side-bar__tags"
        >
          <li
            className="category-side-bar__tag"
          >
            random
             
            <span>
              2
            </span>
          </li>
          <li
            className="category-side-bar__tag"
          >
            test
             
            <span>
              2
            </span>
          </li>
        </ul>
      </div>
      <div
        className="column pt0"
      >
        <div
          className="columns is-multiline"
        >
          <Browser
            author="David"
            github_url="https://www.github.com/boyney123/testing"
            image={Object {}}
            path="/random"
            subtitle="Random item on the website"
            title="Random"
          />
          <Browser
            author="David"
            github_url="https://www.github.com/boyney123/testing"
            image={Object {}}
            path="/random-2"
            subtitle="Another Random item on the website"
            title="New Title 2"
          />
        </div>
        <nav
          aria-label="pagination"
          className="pagination"
          role="navigation"
        >
          <mockConstructor
            className="pagination-previous"
            to="/random"
          >
            ← Previous Page
          </mockConstructor>
          <mockConstructor
            className="pagination-next"
            to="/random/3"
          >
            Next Page →
          </mockConstructor>
          <ul
            className="pagination-list"
          >
            <li>
              <mockConstructor
                aria-label="Goto page 1"
                className="pagination-link"
                to="/random"
              >
                1
              </mockConstructor>
            </li>
            <li>
              <mockConstructor
                aria-label="Goto page 2"
                className="pagination-link is-current"
                to="/random/2"
              >
                2
              </mockConstructor>
            </li>
            <li>
              <mockConstructor
                aria-label="Goto page 3"
                className="pagination-link"
                to="/random/3"
              >
                3
              </mockConstructor>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</Layout>
`;
